src/full/Agda/Interaction/Highlighting/Common.hs

```diff
@@ -3,14 +3,38 @@
 module Agda.Interaction.Highlighting.Common
   ( toAtoms
   , chooseHighlightingMethod
+  , highlightOnlyCode
+  , highlightedFileExt
   ) where
 
+import Agda.Interaction.Options
 import Agda.Interaction.Highlighting.Precise
 import Agda.Syntax.Common
 import Agda.TypeChecking.Monad (HighlightingMethod(..))
 import Data.Maybe (maybeToList)
 import Data.Char (toLower)
 
+-- | Determine how to highlight the file
+highlightOnlyCode :: HtmlHighlight -> FileType -> Bool
+highlightOnlyCode HighlightAll  _ = False
+highlightOnlyCode HighlightCode _ = True
+highlightOnlyCode HighlightAuto AgdaFileType = False
+highlightOnlyCode HighlightAuto MdFileType   = True
+highlightOnlyCode HighlightAuto RstFileType  = True
```

Comment:
```suggestion
highlightOnlyCode HighlightAuto RstFileType  = False
```
Embedding HTML in an `rst` document properly requires som extra work; see:

https://stackoverflow.com/questions/8621042/how-to-embed-html-in-restructured-text-file

It seems that if we enable code-only highlighting automatically for rst files, we are somehow saying that we expect it to work properly.

Reply:
#3373 this should be a separated task